exporters/trace/jaeger/env_test.go

```diff
@@ -0,0 +1,398 @@
+			str:      "1E23",
+			expected: value.Float64(1e23),
+		},
+		{
+			name:     "string: foo",
+			str:      "foo",
+			expected: value.String("foo"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			v := parseValue(tc.str)
+			assert.Equal(t, tc.expected, v)
+		})
+	}
+}
+
+func TestNewRawExporterWithEnv(t *testing.T) {
+	const (
+		collectorEndpoint = "http://localhost"
+		username          = "user"
+		password          = "password"
+		serviceName       = "test-service"
+		disabled          = "false"
+		tags              = "key=value"
+	)
+
+	require.NoError(t, os.Setenv(envEndpoint, collectorEndpoint))
+	require.NoError(t, os.Setenv(envUser, username))
+	require.NoError(t, os.Setenv(envPassword, password))
+	require.NoError(t, os.Setenv(envDisabled, disabled))
+	require.NoError(t, os.Setenv(envServiceName, serviceName))
+	require.NoError(t, os.Setenv(envTags, tags))
+	defer func() {
+		require.NoError(t, os.Unsetenv(envEndpoint))
+		require.NoError(t, os.Unsetenv(envUser))
+		require.NoError(t, os.Unsetenv(envPassword))
+		require.NoError(t, os.Unsetenv(envDisabled))
+		require.NoError(t, os.Unsetenv(envServiceName))
+		require.NoError(t, os.Unsetenv(envTags))
+	}()
+
+	// Create Jaeger Exporter with environment variables
+	exp, err := NewRawExporter(
+		WithCollectorEndpoint(CollectorEndpointFromEnv(), WithCollectorEndpointOptionFromEnv()),
+		WithDisabled(true),
+		WithDisabledFromEnv(),
+		WithProcessFromEnv(),
+	)
+
+	assert.NoError(t, err)
+	assert.Equal(t, false, exp.o.Disabled)
+	assert.EqualValues(t, serviceName, exp.process.ServiceName)
+	assert.Len(t, exp.process.Tags, 1)
+
+	require.IsType(t, &collectorUploader{}, exp.uploader)
+	uploader := exp.uploader.(*collectorUploader)
+	assert.Equal(t, collectorEndpoint, uploader.endpoint)
+	assert.Equal(t, username, uploader.username)
+	assert.Equal(t, password, uploader.password)
+}
+
+func TestNewRawExporterWithEnvImplicitly(t *testing.T) {
+	const (
+		collectorEndpoint = "http://localhost"
+		username          = "user"
+		password          = "password"
+		serviceName       = "test-service"
+		disabled          = "false"
+		tags              = "key=value"
+	)
+
+	require.NoError(t, os.Setenv(envEndpoint, collectorEndpoint))
+	require.NoError(t, os.Setenv(envUser, username))
+	require.NoError(t, os.Setenv(envPassword, password))
+	require.NoError(t, os.Setenv(envDisabled, disabled))
+	require.NoError(t, os.Setenv(envServiceName, serviceName))
+	require.NoError(t, os.Setenv(envTags, tags))
+	defer func() {
+		require.NoError(t, os.Unsetenv(envEndpoint))
+		require.NoError(t, os.Unsetenv(envUser))
+		require.NoError(t, os.Unsetenv(envPassword))
+		require.NoError(t, os.Unsetenv(envDisabled))
+		require.NoError(t, os.Unsetenv(envServiceName))
+		require.NoError(t, os.Unsetenv(envTags))
+	}()
+
+	// Create Jaeger Exporter with environment variables
+	exp, err := NewRawExporter(
+		WithCollectorEndpoint(""),
+		WithDisabled(true),
```

Comment:
I see the current code requires the end-user to always include `WithCollectorEndpoint("")` to ensure the pick up of the collector endpoint from the environment. 

Was this a deliberate decision as opposed to making the collector endpoint more like the `WithDisabledFromEnv(), WithProcessFromEnv()` in that they are auto-added right at the beginning of `NewRawExporter()`?

Reply:
`WithCollectorEndpoint()` overwrites the collector endpoint if the end-user sets up the environment variable.

It looks like the unit test is confused since it uses `""` as input. I can change that input to make it clear.

For example: 

```go
exp, err := NewRawExporter(
	WithCollectorEndpoint("should be overwritten"),
)
```